File: sol/2016/2016.py
class Solution:
    def maximumDifference(self, nums: List[int]) -> int:
        res = -1
        mini = nums[0]
        for x in nums[1:]:
            if x <= mini:
                mini = x
            else:
                res = max(res, x - mini)
        return res

File: sol/2016/2016.cpp
struct Solution {
    int maximumDifference(vector<int>& nums) {
        int res = -1, mini = nums[0], n = nums.size();
        for (int i = 1; i < n; ++i){
            if (nums[i] <= mini){
                mini = nums[i];
            } else {
                res = max(res, nums[i] - mini);
            }
        }
        return res;
    }
};
